src/osd/OSD.cc

```diff
@@ -3942,7 +3943,34 @@ bool OSD::heartbeat_reset(Connection *con)
   return true;
 }
 
+bool OSD::network_rebind_and_check()
+{
+  network_state = 0;
+  set<int> avoid_ports;
+  avoid_ports.insert(cluster_messenger->get_myaddr().get_port());
+  avoid_ports.insert(hb_back_server_messenger->get_myaddr().get_port());
+  avoid_ports.insert(hb_front_server_messenger->get_myaddr().get_port()); 
+
+  bool rebind_ok = false;
+  if (!cluster_messenger->rebind(avoid_ports))
+    if (!cluster_messenger->rebind(avoid_ports))
+      if (!cluster_messenger->rebind(avoid_ports))
+	rebind_ok = true;
+  if (!rebind_ok)
+    network_state |= NETWORK_NEED_REBIND;
 
+  if (cct->_conf->osd_support_ethtool_link_check){
+    bool is_linked;
+    if (cluster_messenger->is_linked())
+      if (!cluster_messenger->is_linked())
+	if (!cluster_messenger->is_linked())
+	  is_linked = true;
```

Comment:
sorry for not getting into it earlier. 
the intention of is_linked is:
0 -- we can sure the ether is unlinked
otherwise -- we cant sure

thanks for your proposals, update!